hyperspy/io_plugins/emd.py

```diff
@@ -377,21 +393,32 @@ def file_reader(filename, load_to_memory=True, log_info=False, **kwds):
 def file_writer(filename, signal, signal_metadata=None, user=None,
                 microscope=None, sample=None, comments=None, **kwds):
     if user is None:  # If not provided, look in metadata:
-        user = signal.metadata.General.as_dictionary().get('user')
-    if user is None:  # If not found, check original_metadata:
-        user = signal.original_metadata.General.as_dictionary().get('user')
+        if 'user' in signal.metadata.General.as_dictionary():
+            user = signal.metadata.General.as_dictionary().get('user')
```

Comment:
A couple of things with this `as_dictionary()` play:
  - `DictionaryTreeBrowser` supports lookup, so you can do `if 'user' in signal.metadata.General` (though it's not necessary in this case).
  - The dictionary is constructed an unnecessary number of times.... It could be done once for `metadata`, once for `original_metadata`, and then just looked up.
  - I would use syntax like this:
```python
metadata = signal.metadata.General.as_dictionary()
user = user or metadata.get('user', None)
microscope = microscope or metadata.get('microscope', None)
sample = sample or metadata.get('sample', None)
# etc
```
You could add the `original_metadata` check in the same one line as well, it's up to you.
```python
orig_metadata = signal.original_metadata.as_dictionary().get('General', None)
user = user or metadata.get('user', None) or (orig_metadata.get('user', None) if orig_metadata else None)
```